engine/src/main/java/org/terasology/engine/utilities/OS.java

```diff
@@ -13,14 +15,23 @@
     MACOSX,
     LINUX;
 
+    public static final String OS_NAME = System.getProperty("os.name").toLowerCase();
+    public static final String ARCHITECTURE = System.getProperty("os.arch");
+    public static final String USER_DIRECTORY = System.getProperty("user.dir");
+    public static final String PATH_SEPARATOR = FileSystems.getDefault().getSeparator();
```

Comment:
I don't like that we have to expose this here 🧐 Usually we would use `Paths` to construct the path in a platform-agnostic way. Is this not possible in `OpenVrProvider.java` due to module sandboxing?